### DISCUSSION TITLE
Feedback

### DISCUSSION THREAD

**Discussion by kelliedesigner** [ministryofjustice/moj-frontend#247] [EXPERIMENTAL: Components, pages and patterns]
# Existing

- [Phase banner](https://design-system.service.gov.uk/components/phase-banner/) on the GOV.UK Design System
- New feedback banner from GDS
![image (2)](https://user-images.githubusercontent.com/6122118/129913994-b1f7402f-6d01-46d7-af07-8ac442b77669.png)

# Variation

### Digital Prison Service

![image (3)](https://user-images.githubusercontent.com/6122118/129914123-61c21bd8-8400-4a78-8de4-c58100e3a32d.png)

**Comment by JSJohal** [comment]
We've re-created the Digital Prison Services feedback banner as shown above into our GOV.UK prototype.

Here is the HTML code for the pattern we used:

```
<div class="feedback-banner" data-test="feedback-banner">
  <a href="https://eu.surveymonkey.com/r/GYB8Y9Q?source=digital-dev.prison.service.justice.gov.uk/" class="govuk-link govuk-link--inverse" target="_blank" rel="noopener noreferrer">
    Give feedback on Digital Prison Services (opens in a new tab)
  </a>
</div>
```

Here is the CSS code for the pattern styling:

`.feedback-banner {
    background: #1d70b8;
    padding: 20px;
    width: 100%;
    max-width: 1170px;
    margin: 0 auto
}`

**Comment by EveBayram** [comment]
## Existing
[Phase banner](https://design-system.service.gov.uk/components/phase-banner/) on the GOV.UK Design System
[Confirmation pages](https://design-system.service.gov.uk/patterns/confirmation-pages/) on the GOV.UK Design System
New feedback banner from GDS (see Kellie's post above)
DPS satisfaction survey (see Kellie's post above)

## Beta feedback (DPS, prisoner education, skills, work and employment)
### Product created in

- GOV.UK prototype kit
- Kotlin (live)

### About the product
The prisoner education, skills, work and employment (ESWE) team is adding more information about individuals to the existing prisoner profile pages on Digital Prison Services (DPS).  

These are non-transactional pages that display information used to assist with multiple tasks. 

DPS already has a feedback mechanism on all pages: a generic satisfaction survey. 

### Who the users are
The new pages will be available to all staff who already have access to DPS inside prisons. This includes staff from a variety of disciplines and areas, such as wing staff, activities hub, Offender Management Unit (OMU), and education department. 

We think that the new information that is being added will be particularly useful for Prison Offender Managers, Key workers, and Activities hub staff. 

### User need
As a member of staff looking at information about a prisoner,
I need to share feedback about what I am seeing (for example, if there is a problem with the page or I think it could be improved), 
So that I can help improve the page for myself and others.

As a member of the digital team responsible for displaying information about a prisoner,
I need to understand how the page is being received and whether it can be improved,
So that I can deliver the best possible solution to our end-users. 

As a member of the digital team responsible for displaying information about a prisoner,
I need to understand which roles are using the page,
So that I can learn more about what this information is used for. 

### The problem we were trying to solve
When we release new non-transactional features on DPS, we need a way of capturing user feedback.

This mechanism could be used in the short-medium term and allow teams to learn from the release so that we can iterate specific new features/ pages during beta.

### How I started
All pages on DPS already include feedback capture. This existing banner goes to a satisfaction survey. This meets the need of getting general feedback on DPS, but does not allow us to capture feedback on specific new, non-transactional features / pages. 

I began by exploring options around modifying the existing DPS satisfaction survey link to suit our purpose. Unfortunately, we were advised that the existing banner could not be changed or removed on a page by page basis. So we needed to do something different. 

### Our particular challenge for DPS pages
Adding a new feedback link is a design challenge due to DPS pages essentially being set out as:

- Header including user details and sign out link
- Comms banner (situational)
- Location and prison switcher
- Breadcrumb navigation
- Body of page
- Satisfaction survey banner
- Footer

Any new feedback link needs to fit within this structure, be findable to the user, and make sense within the hierarchy of this structure / or the page body.

### Review of existing design patterns for feedback capture
<img src="https://user-images.githubusercontent.com/43846343/148804615-1cb87211-dd8c-40a9-9082-3720d801992a.jpg" width="70%" alt="Existing design patterns - feedback component">

I knew that lots of other services have ways of asking for feedback.

I reviewed examples and also looked at a previous review done by the user research community. 

I used this exploration to think about what options might work best for our pages, and started sketching ideas.

### Design crit
I presented some sketches to interaction and content designers in the prison space. This was an opportunity for early feedback, and to check whether this problem had already been solved in DPS. 

The feedback on the sketches helped us narrow the list down to two preferred options. 

![Notes from design critique](https://user-images.githubusercontent.com/43846343/148803802-1caf7a9c-b978-4949-9b75-8555a9b9afcc.png)

### What we decided on
We decided to add a new component to the bottom of the page.

![Original design - beta feedback component](https://user-images.githubusercontent.com/43846343/148803743-f43a37db-b85f-4a2f-95c2-33eddcc38682.png)

We used a section break line to make the component visually distinct from the rest of the page and the existing satisfaction survey banner. 

We thought this design worked well because:

- it's more natural for a user to give feedback after they've had time to read through the page, and our new pages were information heavy

- we know that there is a problem with banner blindness and stacking components with the service header (we found this with the probation service 'service user header' component)

- this feedback component is visually distinct from the existing DPS satisfaction survey banner, and we will monitor whether proximity causes any issues in user testing

### Concerns about the design
Our new screens launched at the start of November 2021.

We had very few responses to the feedback surveys. In 1 month, we received only 3 responses to our new survey plus 3 responses mentioning our new pages within the generic DPS satisfaction survey. This is consistent with the level of responses typically received by the DPS satisfaction survey, but we wanted to learn more.

We decided to revisit the design to see if it would make a difference to the response rate.

### The latest design
We created a new component below the prison switcher in the header area. This is more in line with the existing GOV.UK phase banner.

_(Note: on DPS, the prison switcher does not display a phase badge in the live environment)._

![Current design - beta feedback component](https://user-images.githubusercontent.com/43846343/148803638-d2bbed8b-8c48-4362-a9d1-88349e838cd4.png)

By moving the component to the top of the page we were also trying to make it more distinct from the existing DPS satisfaction survey.

This new design worked better for a page we were building that used a side navigation. Because the feedback component sits above the page body it could accommodate split column page layouts better.

### What is next?

We will continue to monitor the response rate and we hope to learn more from other services who adopt this design or try something different.

### Example of the component in context:
![Beta feedback component in context](https://user-images.githubusercontent.com/43846343/148803676-765674fd-6315-4306-93a4-cd11085aff7e.png)

**Comment by helennickols** [comment]
The feedback banner has been published on the MOJ Design System. This is to make it easier to find and use.



https://design-patterns.service.justice.gov.uk/components/feedback-banner/

**Comment by jrmhaig** [comment]
Are you tracking interest in this component? We have the GovUK phase banner and are changing it to a feedback banner as outlined above.

https://github.com/ministryofjustice/Claim-for-Crown-Court-Defence/pull/9131
